src/Json.Schema.ToDotNet.UnitTests/DataModelGeneratorTests.cs

```diff
@@ -3680,6 +3680,42 @@ public sealed class C
             actual.Should().Be(Expected);
         }
 
+        [Fact(DisplayName = "DataModelGenerator generates virtual properties when option is set")]
+        public void GeneratesVirtualPropertiesWhenOptionIsSet()
+        {
+            _settings.VirtualProperties = true;
+
+            var generator = new DataModelGenerator(_settings, _testFileSystem.FileSystem);
+
+            JsonSchema schema = SchemaReader.ReadSchema(
+@"{
```

Comment:
>@" [](start = 0, length = 2)

We've talked alot about using file-based tests for JSchema. Just a reminder that I'd love to do it, but it's a little tricky in JSchema because the tests can have varying numbers of associated files. Not today.